lib/collector_demon.rb

```diff
@@ -8,21 +8,23 @@ def self.prefix
   end
 
   def run
-    if @pid = fork
-      write_pid_file
-      return
-    end
+    @pid =
+      fork do
+        collector = File.expand_path("../../bin/collector", __FILE__)
 
-    collector = File.expand_path("../../bin/collector", __FILE__)
+        ENV["RUBY_GLOBAL_METHOD_CACHE_SIZE"] = "2048"
+        ENV["RUBY_GC_HEAP_INIT_SLOTS"] = "10000"
+        ENV["PROMETHEUS_EXPORTER_VERSION"] = PrometheusExporter::VERSION
 
-    ENV["RUBY_GLOBAL_METHOD_CACHE_SIZE"] = "2048"
-    ENV["RUBY_GC_HEAP_INIT_SLOTS"] = "10000"
-    ENV["PROMETHEUS_EXPORTER_VERSION"] = PrometheusExporter::VERSION
+        exec collector,
+             GlobalSetting.prometheus_collector_port.to_s,
+             GlobalSetting.prometheus_webserver_bind,
+             parent_pid.to_s,
+             pid_file
+      end
 
-    exec collector,
-         GlobalSetting.prometheus_collector_port.to_s,
-         GlobalSetting.prometheus_webserver_bind,
-         parent_pid.to_s,
-         pid_file
+    Process.detach(@pid)
```

Comment:
> To avoid zombie processes, the parent process should call either:
> 
> [Process.wait](https://docs.ruby-lang.org/en/3.3/Process.html#method-c-wait), to collect the termination > statuses of its children.
>
> [Process.detach](https://docs.ruby-lang.org/en/3.3/Process.html#method-c-detach), to register disinterest in their status.

Calling `Process.detach` here since we don't really care about the child status but it helps prevent zombie processes.